config/routes.rb

```diff
@@ -91,7 +91,13 @@
     end
   end
 
-  resources :submission_assets, only: [:show, :destroy]
+  resources :submission_assets, only: [:show, :destroy] do
+    member do
+      get "rename", controller: :submission_assets, as: "rename"
+    end
+  end
+  patch '/submission_assets/:id', to: 'submission_assets#update'
```

Comment:
You can simply add this route by enabling the `#update` action on the `submission_assets` resource. 

```suggestion
  resources :submission_assets, only: [:show, :update, :destroy] do
    member do
      get "rename", controller: :submission_assets, as: "rename"
    end
  end
```